lib/path.js

```diff
@@ -1172,7 +1173,7 @@ const posix = {
     let joined;
     for (let i = 0; i < args.length; ++i) {
       const arg = args[i];
-      validateString(arg, 'path');
+      validateDataTypes(arg, 'path', ['Buffer', 'String']);
```

Comment:
We should not make any change to `node:path`, its API should be considered stable at this point.

Reply:
In that case,

modifying the line to `pathModule.join(path.toString(), result[i].toString()),` in would be the easy option. But, it will convert the buffer to string. 

```
processReadResult(path, result) {
    for (let i = 0; i < result.length; i += 2) {
      ArrayPrototypePush(
        this[kDirBufferedEntries],
        getDirent(
          pathModule.join(path, result[i]),
          result[i],
          result[i + 1],
        ),
      );
    }
  }
  ```

Do we need to retain the Buffer type and return the directories in Buffer instead of string?

Reply:
We do need to keep using buffer, as we can’t assume what’s the encoding.

Reply:
Okay. For that, I can write a function which can use `Buffer.concat(paths); ` instead of `path.join`.

As you adviced, we can untouch path module and write this one inside dir module itself. 

Please let me know your thoughts.

Reply:
Fixed.